core/src/main/java/hudson/tasks/ArtifactArchiver.java

```diff
@@ -233,6 +251,19 @@ public void perform(Run<?,?> build, FilePath ws, Launcher launcher, TaskListener
 
             Map<String,String> files = ws.act(new ListFiles(artifacts, excludes, defaultExcludes, caseSensitive));
             if (!files.isEmpty()) {
+                if (isFlattenDirectories()) {
+                    Map<String,String> flattenedFiles = new LinkedHashMap<>();
+                    for (Map.Entry<String,String> soloFile : files.entrySet()) {
+                        String simpleFile = soloFile.getKey().replaceFirst(".+/", "");
```

Comment:
surely `.*/`  - you may be trying to archive directly from root (lord knows why but `/bob` should map to `bob` and not `/bob`)....